source/postprocess/visualization/shear_stress_eigenvectors.cc

```diff
@@ -0,0 +1,340 @@
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
+              names.emplace_back("shear_stress_eigenvectors_sigma_2");
+              names.emplace_back("shear_stress_eigenvectors_sigma_2");
+              names.emplace_back("shear_stress_eigenvectors_sigma_2");
+              names.emplace_back("shear_stress_eigenvalue_sigma_1");
+              names.emplace_back("shear_stress_eigenvalue_sigma_2");
+              break;
+
+            case 3:
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
+              names.emplace_back("shear_stress_eigenvectors_sigma_2");
+              names.emplace_back("shear_stress_eigenvectors_sigma_2");
+              names.emplace_back("shear_stress_eigenvectors_sigma_2");
+              names.emplace_back("shear_stress_eigenvectors_sigma_3");
+              names.emplace_back("shear_stress_eigenvectors_sigma_3");
+              names.emplace_back("shear_stress_eigenvectors_sigma_3");
+              names.emplace_back("shear_stress_eigenvalue_sigma_1");
+              names.emplace_back("shear_stress_eigenvalue_sigma_2");
+              names.emplace_back("shear_stress_eigenvalue_sigma_3");
+              break;
+
+            default:
+              Assert (false, ExcNotImplemented());
+          }
+        return names;
+      }
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      ShearStressEigenvectors<dim>::get_data_component_interpretation () const
+      {
+        std::vector<DataComponentInterpretation::DataComponentInterpretation> components
+        (dim*dim,DataComponentInterpretation::component_is_part_of_vector);
+        components.emplace_back(DataComponentInterpretation::component_is_scalar);
+        components.emplace_back(DataComponentInterpretation::component_is_scalar);
+        components.emplace_back(DataComponentInterpretation::component_is_scalar);
+        return components;
+      }
+
+
+
+      template <int dim>
+      UpdateFlags
+      ShearStressEigenvectors<dim>::get_needed_update_flags () const
+      {
+        return update_gradients | update_values | update_q_points;
+      }
+
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(ShearStressEigenvectors,
+                                                  "shear stress eigenvectors",
+                                                  "A plugin that computes the direction and magnitude of the "
+                                                  "maximum horizontal component of the compressive stress as a vector "
```

Comment:
You'll have to go through the documentation here and update it. I think this is copied from the other class?

It would probably also be good to add a sentence to both this and the other plugin, referencing each other.